core/src/main/kotlin/entity/Asset.kt

```diff
@@ -0,0 +1,90 @@
+package dev.kord.core.entity
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.KordObject
+import dev.kord.rest.Image
+import dev.kord.rest.route.CdnUrl
+import dev.kord.rest.route.DiscordCdn
+
+public sealed class Asset(
```

Comment:
shouldn't this instead be a normal final class with factory functions in the companion object? i don't see any values in the subclasses.

so instead of
```kt
public class DefaultUserAvatar(discriminator: Int, kord: Kord) :
    Asset(Image.Format.PNG, DiscordCdn.defaultAvatar(discriminator), kord)
```
we would have
```kotlin
public companion object {
    public fun defaultUserAvatar(discriminator: Int, kord: Kord): Asset =
        Asset(Image.Format.PNG, DiscordCdn.defaultAvatar(discriminator), kord)
}
```